Objects/exceptions.c

```diff
@@ -216,6 +219,34 @@ BaseException_set_args(PyBaseExceptionObject *self, PyObject *val, void *Py_UNUS
     return 0;
 }
 
+static PyObject *
+BaseException_get_note(PyBaseExceptionObject *self, void *Py_UNUSED(ignored))
+{
+    if (self->note == NULL) {
+        Py_RETURN_NONE;
+    }
+    Py_INCREF(self->note);
+    return self->note;
+}
+
+static int
+BaseException_set_note(PyBaseExceptionObject *self, PyObject *note, void *Py_UNUSED(ignored))
+{
+    if (note == NULL) {
+        PyErr_SetString(PyExc_TypeError, "__note__ may not be deleted");
+        return -1;
+    }
+    else if (note != Py_None && !PyUnicode_Check(note)) {
+        PyErr_SetString(PyExc_TypeError,
+                        "__note__ must be an str or None");
```

Comment:
PEP 7 (fits in 79 chars)

```suggestion
        PyErr_SetString(PyExc_TypeError, "__note__ must be an str or None");
```